### DISCUSSION TITLE
Debian under Microsoft WSL. Ghostty appears to crash on exit (sometimes)

### DISCUSSION THREAD

**Discussion by DudleyFox** [ghostty-org/ghostty#2952] [Feature Requests, Ideas] [answered]
Ghostty appears to crash on exit sometimes. I'm running debian under WSL on Microsoft windows 11. I have uploaded the crash reports using the sentry CLI.

This is not really an issue for me, since it runs fine. I only see it on shutdown.

Running a build from source using the recommend:
zig build -p $HOME/.local -Doptimize=ReleaseFast

# Reproduction steps

1. Run ghostty from the windows debian shell
2. Let ghostty load (fast!)
3. type `exit` and hit enter

I have made it crash more reliably if I do the following:
1. Run ghostty from the windows debian shell
2. Let ghostty load
3. Open a new ghostty window with Ctrl-Shift-N
4. open and close neovim in window 1
5. open and close neovim in window 2
6. type `exit` and hit enter

# System Info
_uname -a_
Linux anonhost 5.15.167.4-microsoft-standard-WSL2 #1 SMP Tue Nov 5 00:21:55 UTC 2024 x86_64 GNU/Linux

_systeminfo_
Host Name:                 ANONHOST
OS Name:                   Microsoft Windows 11 Home
OS Version:                10.0.22631 N/A Build 22631
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          redacted@redacted
Registered Organization:   N/A
Product ID:                00342-22159-42311-AAOEM
Original Install Date:     8/27/2023, 6:33:06 AM
System Boot Time:          12/11/2024, 5:14:30 PM
System Manufacturer:       ASUSTeK COMPUTER INC.
System Model:              ASUS TUF Gaming A17 FA707NU_FA707NU
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: AMD64 Family 25 Model 68 Stepping 1 AuthenticAMD ~3201 Mhz
BIOS Version:              American Megatrends International, LLC. FA707NU.315, 10/18/2024
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     64,763 MB
Available Physical Memory: 53,432 MB
Virtual Memory: Max Size:  68,859 MB
Virtual Memory: Available: 56,498 MB
Virtual Memory: In Use:    12,361 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\ANONHOST
Hotfix(s):                 7 Hotfix(s) Installed.
                           [01]: KB5045935
                           [02]: KB5012170
                           [03]: KB5026039
                           [04]: KB5027397
                           [05]: KB5029517
                           [06]: KB5048685
                           [07]: KB5046729
Network Card(s):           3 NIC(s) Installed.
                           [01]: Realtek 8852BE Wireless LAN WiFi 6 PCI-E NIC
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     172.17.2.1
                                 IP address(es)
                                 [01]: 172.17.2.142
                                 [02]: fe80::3118:bf7a:825d:a201
                                 [03]: 2600:1700:e950:37bf:5ee:50ac:8411:c13a
                                 [04]: 2600:1700:e950:37bf:a112:1b47:7287:2cfd
                                 [05]: 2600:1700:e950:37bf::15ad
                           [02]: Realtek PCIe GbE Family Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [03]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.

**Comment by mitchellh** [comment, marked as answer]
WSL isn't really a supported target currently. It isn't tested and I don't have a dev environment to work on issues for it. I'm happy to fix this if you can find the root cause (or anyone else) or if we can reproduce it on a supported platform.

**Reply by DudleyFox** [reply]
I figured WSL was an edge case, but wanted to report it in case it was useful. If I have time I'll see if I can find the root cause, but I am very new to Zig.

**Reply by DudleyFox** [reply]
This is the call stack from the logs running the debug version:

```
General protection exception (no address available)
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/notebook.zig:342:52: 0x326107e in closeTab (ghostty)
                    c.gtk_window_destroy(tab.window.window);
                                                   ^
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/Window.zig:432:27: 0x31b5e8d in closeTab (ghostty)
    self.notebook.closeTab(tab);
                          ^
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/Tab.zig:121:25: 0x31b5edb in remove (ghostty)
    self.window.closeTab(self);
                        ^
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/Surface.zig:207:34: 0x31515e3 in remove (ghostty)
            .tab_ => |t| t.remove(),
                                 ^
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/Surface.zig:700:30: 0x30de8da in close (ghostty)
        self.container.remove();
                             ^
/home/user007/github/ghostty-org/ghostty/src/Surface.zig:727:26: 0x30dcd94 in close (ghostty)
    self.rt_surface.close(self.needsConfirmQuit());
                         ^
/home/user007/github/ghostty-org/ghostty/src/Surface.zig:915:23: 0x30dc284 in handleMessage (ghostty)
            self.close();
                      ^
/home/user007/github/ghostty-org/ghostty/src/App.zig:486:34: 0x30ddb95 in surfaceMessage (ghostty)
        try surface.handleMessage(msg);
                                 ^
/home/user007/github/ghostty-org/ghostty/src/App.zig:265:62: 0x30dde95 in drainMailbox (ghostty)
            .surface_message => |msg| try self.surfaceMessage(msg.surface, msg.message),
                                                             ^
/home/user007/github/ghostty-org/ghostty/src/App.zig:144:26: 0x30de5d6 in tick (ghostty)
    try self.drainMailbox(rt_app);
                         ^
/home/user007/github/ghostty-org/ghostty/src/apprt/gtk/App.zig:1094:51: 0x30decef in run (ghostty)
        const should_quit = try self.core_app.tick(self);
                                                  ^
/home/user007/github/ghostty-org/ghostty/src/main_ghostty.zig:111:24: 0x30e02d5 in main (ghostty)
    try app_runtime.run();
                       ^
/home/user007/bin/zig/lib/std/start.zig:524:37: 0x30e0d3e in main (ghostty)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x7fc695574249 in ??? (libc.so.6)
Unwind information for `libc.so.6:0x7fc695574249` was not available, trace may be incomplete


```
